src/index.js

```diff
@@ -45,13 +48,213 @@ module.exports = {
+           for (let i=0; i<currW.length; i += 1) {
+             varXLocal.push(varXLocalMatrix[i][i]);
+          }  
+
+          /* eslint-disable no-console */
+           console.log('local r squared for currW', rSquaredLocal);
+           console.log('local t Values for currW', tValueLocal);
+           console.log('local p Values for currW', pValueLocal);
+          /* eslint-enable no-console */
+
+           return {
+             sseLocal,
+             sstLocal,
+             varXLocal,
+             currW,
+             localCount,
+             rSquared,
+             tValue,
+             pValue,
+             rSquaredLocal,
+             tValueLocal,
+             pValueLocal,
+             endOfIteration : true,
+             statisticStep : 1,
+          };
+        }
+        else {
+        return localRunner.run(opts);        
+       }
+      },
       verbose: true,
     },
   ],
   remote: {
     type: 'function',
-    fn(opts) { return remoteRunner.run(opts); },
+    fn(opts) { 
+     
+     // Extract X and y variate name from userData
+     const X_label=Object.keys(opts.userResults[0].userData.files[0].tags);
+     const y_label=opts.userResults[0].userData.__DECLARATION_INPUTS_KEY__[0][0];
+     
+     if ((opts.userResults[0].data.endOfIteration == true) && (opts.userResults[0].data.statisticStep == 0)){
+         // if statisticStep==0, remote node calculates the globalMeanY and return it to local node
+         const userResults = opts.userResults;
+
+         // Not all user results contain betas. Return early.
+         if (userResults.some(userResult => !((userResult || {}).data || {}).betaVector)) {
+            return {};
+           }
+
+         // calculate averageBetaVector
+         const currW = opts.previousData.currW;
+
+         // calculate globalMeanY
+         const siteCount=userResults.length;
+         var totalY=0;
+         var globalYCount=0;
+
+         for (let i=0; i < siteCount; i += 1) {
+            totalY += userResults[i].data.localMeanY*userResults[i].data.localCount;
+            globalYCount += userResults[i].data.localCount;
+         }
+
+         const globalMeanY = totalY/globalYCount;
+
+         /* eslint-disable no-console */
+         console.log('currW :', currW);
+         console.log('globalMeanY is :', globalMeanY);
+         /* eslint-enable no-console */
+   
+           
+ 
+         return { currW,
+                 globalMeanY,
+                 endOfIteration : true,
+                 statisticStep : 1,
+                };
+
+     }
+    else if ((opts.userResults[0].data.endOfIteration == true) && (opts.userResults[0].data.statisticStep == 1)) {
+         // if statisticStep==1, remote node calculates the global statistics 
+         
+         const userResults = opts.userResults;
+
+        // get passed parameters from local nodes
+         const currW = opts.previousData.currW;
+         const betaVectorLocal = userResults.map(r => r.data.betaVector);
```

Comment:
There’s a lot of repeated iteration over `userResults`, [similar to _decentralized-single-shot-ridge-regression_](https://github.com/MRN-Code/decentralized-single-shot-ridge-regression/pull/11#discussion_r96543959).